externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -168,15 +169,23 @@ object SparkSQLEngine extends Logging {
 
   def main(args: Array[String]): Unit = {
     SignalRegister.registerLogger(logger)
-    var spark: SparkSession = null
-    try {
-      spark = createSpark()
+    val submitTime = kyuubiConf.getOption(KYUUBI_ENGINE_SUBMIT_TIME_KEY) match {
+      case Some(t) => t.toLong
+      case _ => 0L
+    }
+    val initTimeout = kyuubiConf.get(ENGINE_INIT_TIMEOUT)
+    if (System.currentTimeMillis() - submitTime > initTimeout) {
```

Comment:
If we do nothing here, it will exit abnormally. Is that ok?
cc @yaooqinn @turboFei

Reply:
>  it will exit abnormally.

What you mean here is that picture you show in the PR desc?

Reply:
What if we through exception here?